ihe-interop/src/test/java/org/motechproject/ihe/interop/handler/helper/IHEActionHelperTest.java

```diff
@@ -61,4 +69,25 @@ public void shouldThrowExceptionWhenRecipientNotFound() throws ClassNotFoundExce
         when(iheTemplateDataService.findByName("sampleTemplate3")).thenReturn(cdaTemplate);
         iheActionHelper.handleAction(params);
     }
+
+    @Test
+    public void shouldCallSendTemplateMethod() throws IOException, SAXException, ParserConfigurationException, ClassNotFoundException {
+        Map<String, Object> params = new HashMap<>();
+        params.put(Constants.TEMPLATE_NAME_PARAM, "sampleTemplate3");
+        params.put(Constants.RECIPIENT_NAME_PARAM, "sampleRecipient3");
+
+        byte[] bytes = IOUtils.toByteArray(getClass().getResourceAsStream("/empty_template.xml"));
+        Byte[] byteObjects = new Byte[bytes.length];
+        int i = 0;
+        for (byte b : bytes) {
+            byteObjects[i++] = b;
+        }
+
+        when(iheTemplateDataService.findByName("sampleTemplate3")).thenReturn(cdaTemplate);
+        when(hl7RecipientsService.getRecipientbyName("sampleRecipient3")).thenReturn(hl7Recipient);
+        when((Byte[]) iheTemplateDataService.getDetachedField(cdaTemplate, "templateData")).thenReturn(byteObjects);
+        verify(iheTemplateService).sendTemplateToRecipientUrl(new String(bytes, StandardCharsets.UTF_8), new PostMethod(hl7Recipient.getRecipientUrl()));
```

Comment:
This should be called after iheActionHelper.handleAction(params). This should return exception here. I checked this and the problem is with hl7Recipient.getRecipientUrl() - without this, the test throw an exception "Wanted but not invoked".